packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -108,11 +112,21 @@ abstract class Route<T> {
 
   /// The route should remove its overlays and free any other resources.
   ///
-  /// A call to didPop() implies that the Route should call dispose() itself,
-  /// but it is possible for dispose() to be called directly (e.g. if the route
-  /// is replaced, or if the navigator itself is disposed).
+  /// This route is no longer referenced by the navigator.
```

Comment:
maybe say that it should only be called by the navigator? or maybe it's self-evident that dispose is only called by the owner.